### PR TITLE
Docs update and spellchecking

### LINKED ISSUE #119: Update docs to allow Spaceflights tutorial run with 'kedro<0.18' and kedro-kubeflow==0.6.2
`kedro new --starter=spaceflights` doesn't work out of the box on Kubeflow Pipelines and package versions:
```
kedro<0.18
kedro-kubeflow==0.6.2 
```

## Proposed updates:

1) Add kedro-kubeflow in requirements files to solve the problem during onExit node execution:
```Error: no such command 'kubeflow'```

Add: in `requirements.txt` / `requirements.in`
```
kedro-kubeflow==0.6.2 
```

2) Add pipeline outputs to allow save data processing nodes outputs as `pickle.PickleDataSet`

`src/spaceflights/data_processing/pipeline.py`

```
        inputs=["companies", "shuttles", "reviews"],
        ## Add this line
        outputs=["model_input_table","preprocessed_shuttles", "preprocessed_companies"]
    )
```

3) Solve the unsupported `.` character in the node name.

In the latest Spaceflights starter, we have 2 sub-pipelines for the data science part: active_modelling_pipeline and candidate_modelling_pipeline in `src/spaceflights/data_science/pipeline.py`.

When we add pipeline outputs like in 2) above, output names will look like this: "active_modelling_pipeline.X_train" and we can't compile it to Kubeflow Pipeline because of unallowed `.` in the node name added by the namespace.

We can temporarily resolve this issue by removing `candidate_modelling_pipeline` and keeping only one pipeline like this:

`src/spaceflights/data_science/pipeline.py`
```
 ds_pipeline = pipeline(
        pipe=pipeline_instance,
        inputs="model_input_table"
    )
    return pipeline(
        pipe=ds_pipeline, 
        inputs="model_input_table",
        namespace="data_science",
        ## Add this
        outputs=["regressor", "X_train", "X_test", "y_train", "y_test"],
```

but this is rather a workaround than the root cause solution.

### LINKED ISSUE #132: Set up Dependabot and CodeQL
Set up Dependabot and CodeQL scans to enable automated dependency updates and security alerts.

### LINKED ISSUE #155: Add instruction to documentation how to replicate fully test env locally with kind k8s cluster testing.
There are few more quirks to test things locally that could be documented.

### LINKED ISSUE #162: Add spellchecking step to precommit

### PR DESCRIPTION
**Pull request by Lasica**
#### Description

Updating the documentation with extra info about difficulties during the setup with starter kedro and about contributing/local testing. Also adds spellcheck github action,

Resolves #155 
Resolves #162
Resolves #119 
Resolves #132

##### PR Checklist
- [x] [Changelog](CHANGELOG.md) updated

### CODE REVIEW

**Comment by Lasica** [comment]
Question is whether to use the same spellcheck for github action and for pre-commit. There is only one issue as it has external dependencies - it uses aspell or hunspell engines with dictionaries for spellcheck. There is no support in pre-commit for pyspelling tool, there are other spell checkers (`codespell`, `spell-check` and `typos`).

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  spellcheck:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Spellcheck
+      - uses: actions/checkout@v2
+      - uses: igsekor/pyspelling-any@v0.0.2
```
Comment (mwiewior) [inline .github/workflows/spellcheck.yml:21]: imho it's better to use: https://github.com/rojopolis/spellcheck-github-actions

File: .github/workflows/spellcheck.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# This is workflow for spell checking using PySpelling lib (https://pypi.org/project/pyspelling/)
+name: Spellcheck
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    paths: "docs/**"
```
Comment (mwiewior) [inline .github/workflows/spellcheck.yml:7]: imho should be triggered only on PR events

File: docs/source/03_getting_started/01_quickstart.md [hunk 3]
```diff
@@ -54,6 +54,17 @@ Change directory to the project generated in /home/mario/kedro/kubeflow-plugin-d
 A best-practice setup includes initialising git and creating a virtual environment before running `kedro install` to install project-specific dependencies. Refer to the Kedro documentation: https://kedro.readthedocs.io/
 ```
 
+There are some adjustments that need to be made to run starter Kedro project on Kubeflow. We need to replace all dots in names of catalog with other characters as Kubeflow does not accept dots in names. You can do it by using `sed` in starter root directory:
```
Comment (em-pe) [inline docs/source/03_getting_started/01_quickstart.md:57]: This is not true, currently end 2 end tests are passing with catalog entries having dots. What I would mention instead is that the plugin was not yet adjusted to work with [kedro namespaces](https://kedro.readthedocs.io/en/stable/tutorial/namespace_pipelines.html) so using them may cause issues.

**Review by em-pe — CHANGES_REQUESTED** [review]

File: docs/source/03_getting_started/01_quickstart.md [hunk 4]
```diff
@@ -54,6 +54,17 @@ Change directory to the project generated in /home/mario/kedro/kubeflow-plugin-d
...
+```console
+for i in {1..10}; do sed -r 's/^([^ \t])([^. ]*)\./\1\2_/g' conf/base/catalog.yml > conf/base/catalog.yml; done
```
Comment (mwiewior) [inline docs/source/03_getting_started/01_quickstart.md:59]: No sure if got this point?

File: docs/source/03_getting_started/01_quickstart.md [hunk 5]
```diff
@@ -54,6 +54,17 @@ Change directory to the project generated in /home/mario/kedro/kubeflow-plugin-d
...
+```
+
+Before installing the dependencies, add the `kedro-kubeflow` and `kedro-docker` to `requirements.*` in src:
+```console
```
Comment (mwiewior) [inline docs/source/03_getting_started/01_quickstart.md:63]: Why it isn't put there by default and manual action is needed ?

File: docs/source/04_contributing/02_local_testing.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+The kind has its own docker registry that you need to upload the image to. However, since it does not have any connection to other registry we want to prevent it from trying to pull any image. In order to do that, we need to tag the built docker image with specific version.
+
+Locate your image name (it should be the same as kedro project name) with:
+```
+docker images
+```
+
+Then tag your image with the following command (the version is arbitrary and can be any other version):
+```
+docker tag <image>:latest <image>:1.0
```
Comment (mwiewior) [inline docs/source/04_contributing/02_local_testing.md:32]: Why do we need to tag it instead of using `latest` tag

**Review by mwiewior — CHANGES_REQUESTED** [review]

File: docs/source/03_getting_started/01_quickstart.md (hunk 4 above)
Comment (Lasica) [reply docs/source/03_getting_started/01_quickstart.md]: This point has been invalid and was refactored/fixed in new pr.

File: .github/workflows/spellcheck.yml (hunk 2 above)
Comment (Lasica) [reply .github/workflows/spellcheck.yml]: Updated.

File: .github/workflows/spellcheck.yml (hunk 1 above)
Comment (Lasica) [reply .github/workflows/spellcheck.yml]: Updated.

File: docs/source/03_getting_started/01_quickstart.md (hunk 5 above)
Comment (Lasica) [reply docs/source/03_getting_started/01_quickstart.md:68]: I've created a separate issue for that: #161

File: docs/source/04_contributing/01_guideline.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+For linting and keeping code clean we use pre-commit package to join with github hooks. Use it by doing:
+
+```bash
+pip install pre-commit
+pre-commit install
+```
+
+## Releasing new versions
+
+TBD
```
[the remark this answers is not in the document]
Comment (em-pe) [reply docs/source/04_contributing/01_guideline.md:35]: see CONTRIBUTING.md

**Comment by Lasica** [comment]
Now also resolves #162

File: docs/source/03_getting_started/01_quickstart.md (hunk 5 above)
Comment (mwiewior) [reply docs/source/03_getting_started/01_quickstart.md:65]: ok, thx

**Review by em-pe — APPROVED** [review]

File: docs/source/03_getting_started/05_authentication.md [hunk 8]
```diff
@@ -6,13 +6,13 @@ Plugin supports 2 ways of authenticating to Kubeflow Pipelines API:
...
 
-## 2. KFP behind Dex+authservice
+## 2. KFP behind Dex with `authservice`
 
 Dex is the recommended authentication mechanism for on-premise Kubeflow clusters. The usual setup looks in a way that:
 
 * [oidc-autheservice](https://github.com/arrikto/oidc-authservice) redirect unauthenticated users to Dex,
 * [Dex](https://github.com/dexidp/dex) authenticates user in remote system, like LDAP or OpenID and also acts as OpenID provider,
-* oidc-autheservice asks Dex for token and creates the session used across entire Kubeflow.
+* `oidc-autheservice` asks Dex for token and creates the session used across entire Kubeflow.
```
Comment (mwiewior) [inline docs/source/03_getting_started/05_authentication.md:15]: For a token?

File: docs/source/04_contributing/02_local_testing.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
+(local-testing)=
+# Local testing
+
+## Unit tests
+
+The plugin has unit tests that can be run with `tox`:
+```console
+$ pip install tox-pip-version
+$ tox -v -e py38
```
Comment (mwiewior) [inline docs/source/04_contributing/02_local_testing.md:9]: Maybe it should be stated clearly that py38 env is just an example for Python 3.8 not the only option?

File: docs/spellcheck_exceptions.txt [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+CPython
+CShellActivator
+DSL
+Dex
+EmptyDirVolumeSource
+FishActivator
+FromAppData
+GCP
+Github
+IAP
```
Comment (mwiewior) [inline docs/spellcheck_exceptions.txt:12]: Maybe it is possible to do case-insensitive string comparison so not put the same entry multiple times?

**Review by mwiewior — CHANGES_REQUESTED** [review]
Add kedro-mlflow  supported version info

File: docs/spellcheck_exceptions.txt (hunk 10 above)
Comment (Lasica) [reply docs/spellcheck_exceptions.txt:12]: yes, found an option and changed that

File: docs/source/03_getting_started/05_authentication.md (hunk 8 above)
Comment (Lasica) [reply docs/source/03_getting_started/05_authentication.md]: fixed

File: docs/source/04_contributing/02_local_testing.md (hunk 9 above)
Comment (Lasica) [reply docs/source/04_contributing/02_local_testing.md:9]: fixed

**Review by mwiewior — APPROVED** [review]

**Pull request merged**